### PR TITLE
Added compatibility for Bevy 0.12

### PR DESCRIPTION
**Pull request by HarryTheWulf**
This is just a simple thing to update the Cargo.toml to add support for the latest version of Bevy.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "bevy_enum_filter"
-version = "0.2.0"
+version = "0.2.1"
```
Comment (MrGVSV) [inline Cargo.toml]: I believe the version should be `0.3.0` since upgrading Bevy is a breaking change.

**Review by MrGVSV — COMMENTED** [review]
Thanks for doing this! One quick comment

File: Cargo.toml (hunk 1 above)
Comment (HarryTheWulf) [reply Cargo.toml]: Okay, absolutely!
Like the previous merge I wasn't sure whether to do this or not.
I updated the README too so will change that accordingly.

Comment (MrGVSV) [reply Cargo.toml]: Yeah normally I prefer to update the version numbers myself, but I've been super busy IRL and so I don't trust myself to even remember to do something as simple as that lol

Comment (HarryTheWulf) [reply Cargo.toml]: I've updated the version number to `0.3.0` as requested :)

**Review by MrGVSV — APPROVED** [review]
Thanks for doing this! I'll try to publish a release tonight!

**Pull request merged**